x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/engine/Engine.java

```diff
@@ -163,19 +186,24 @@ public String[] indices() {
         return analyticsCollectionName;
     }
 
+    public long updatedAt() {
+        return updatedAt;
+    }
+
     @Override
     public boolean equals(Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
         Engine engine = (Engine) o;
         return name.equals(engine.name)
             && Arrays.equals(indices, engine.indices)
-            && Objects.equals(analyticsCollectionName, engine.analyticsCollectionName);
+            && Objects.equals(analyticsCollectionName, engine.analyticsCollectionName)
+            && updatedAt == engine.updatedAt();
```

Comment:
I wasn't sure if we wanted to included `updatedAt` in `hashCode()` and `equals()`.

Reply:
Lets' keep it yes, it's mainly for tests that we have these functions implemented.